### ISSUE TITLE
S3 Plugin Isn't closing connections to S3

### ISSUE THREAD

**Issue by geofffranks** [starkandwayne/shield#472]
We're running into an issue with shield running a 23GB backup to a Scality-based S3 API. Our target configuration works great for smaller backups, but when we get to a larger backup, it fails with the following error:

```
dial tcp: lookup REDACTED on REDACTED:53: dial udp REDACTED:53: socket: too many open files
```

Upon looking closer at the agent during the backup, the S3 process gets up to it's 1024 open file limit, and then crashes. All these open files are sockets that appear to have been connections to the S3 server. Some were in CLOSE_WAIT with data in the receive queue, others had just timed out in the kernel and had no status/protocol associated with them. 

We've worked around this for now by increasing the multi-part-upload-size to require fewer connections to the backend, but the 23GB is backup is our sandbox environment, and will be substantially larger in our prod env.

**Comment by jiangytcn** [comment]
@geofffranks  May know the 23GB is multiple files or a single object to be backed up ?

**Comment by geofffranks** [comment]
It’s a single stream coming out of bbr from a backup of a CF deployment

**Comment by jhunt** [comment]
@patrick1378 updating vendor deps to latest jhunt/go-s3 should take care of this, but may require some changes to how we create uploads (go will helpfully let you know)

**Comment by patrick1378** [comment]
Should be fixed with 1bc8596b8f4748f527be76c65e5e8eb4fc2a0ffe.
